jupyter_client/kernelspec.py

```diff
@@ -54,19 +56,47 @@ def to_json(self):
         """
         return json.dumps(self.to_dict())
 
+
+_kernel_name_pat = re.compile(r'^[a-z0-9._\-]+$', re.IGNORECASE)
+
+def _valid_kernel_name(name):
+    """Check that a kernel name is valid.
+    
+    Returns True if valid, False if not.
```

Comment:
If you rename this function to `_is_valid_kernel_name` you could get away with not having to explain the return value, and be stylistically consistent with `_is_kernel_dir` below.  A small suggestion, feel free to ignore if you like it better this way.